### PR TITLE
Fixed Weird Tp Fail Damage

### PR DESCRIPTION
**Pull request by CoolPuppyKid**
this fixes https://github.com/cuberite/cuberite/issues/5542

i fixed it by not allowing fall damage while teleporting

### CODE REVIEW

**Comment by CoolPuppyKid** [comment]
btw i have verified in game that this fixes it

**Comment by CoolPuppyKid** [comment]
i fixed clang build lol

**Comment by CoolPuppyKid** [comment]
> #5544 is definitely the better fix, I'll merge this in though and keep the other PR open, once somebody gets around to it we can fix the other one up properly and merge after reverting this change.
> 
> @CoolPuppyKid this looks good, please add your name into the CONTRIBUTORS file so you are recognised as a contributor and accepting the license.

i have added my name to it

File: CONTRIBUTORS [hunk 1]
```diff
@@ -106,3 +106,4 @@ worktycho
 Xenoxis
 xoft (Mattes Dolak/madmaxoft on GH)
 Yeeeeezus (Donated AlchemistVillage prefabs)
+CoolPuppyKid
```
Comment (bearbin) [inline CONTRIBUTORS:109]: This file is alpha-sorted, you need to slot yourself in at the right alphabetical position!

Comment (CoolPuppyKid) [reply CONTRIBUTORS:109]: ok will do

Comment (CoolPuppyKid) [reply CONTRIBUTORS]: > This file is alpha-sorted, you need to slot yourself in at the right alphabetical position!

fixed now

**Review by bearbin — APPROVED** [review]

**Pull request merged**